### PR TITLE
Add `isRecovering`

### PR DESCRIPTION
**Pull request by erikzhang**

### CODE REVIEW

File: neo/Consensus/ConsensusService.cs [hunk 1]
```diff
@@ -154,11 +155,18 @@ private void InitializeConsensus(byte viewNumber)
             if (context.MyIndex == context.PrimaryIndex)
             {
                 context.State |= ConsensusState.Primary;
-                TimeSpan span = TimeProvider.Current.UtcNow - block_received_time;
-                if (span >= Blockchain.TimePerBlock)
-                    ChangeTimer(TimeSpan.Zero);
+                if (isRecovering)
+                {
+                    ChangeTimer(TimeSpan.FromSeconds(Blockchain.SecondsPerBlock << (viewNumber + 1)));
```
Comment (jsolman) [inline neo/Consensus/ConsensusService.cs]: If recovering only restores the `ChangeView` messages and no prepare request, I don't think this is the right behavior. Perhaps #593 is better for that reason.

Comment (erikzhang) [reply neo/Consensus/ConsensusService.cs]: Fixed in 31218da

**Comment by jsolman** [comment]
Closes #593

File: neo/Consensus/ConsensusService.cs (hunk 1 above)
Comment (jsolman) [reply neo/Consensus/ConsensusService.cs]: Nice solution.

**Pull request merged**